File: app/src/main/java/net/itsjustsomedude/tokens/db/Coop.java
package net.itsjustsomedude.tokens.db;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

import java.util.Calendar;
import java.util.List;

//@Entity
//data class User(
//        @PrimaryKey val uid: Int,
//        @ColumnInfo(name = "first_name") val firstName: String?,
//        @ColumnInfo(name = "last_name") val lastName: String?
//)

@Entity
public class Coop {
    @PrimaryKey(autoGenerate = true)
    public long id;

    public String name;
    public String contract;
    public Calendar startTime;
    public Calendar endTime;
    public boolean sinkMode;

    public List<String> players;

    public Coop() {
        this("New Coop", "KevID", null, null, false);
    }

    public Coop(String name, String kevId, Calendar startTime, Calendar endTime, boolean sinkMode) {
        this.name = name;
        this.contract = kevId;
        this.startTime = startTime;
        this.endTime = endTime;
        this.sinkMode = sinkMode;
    }
}

File: app/src/main/java/net/itsjustsomedude/tokens/db/Coop.kt
package net.itsjustsomedude.tokens.db

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.Calendar

@Entity
data class Coop(
	@PrimaryKey(autoGenerate = true)
	val id: Long = 0,

	val name: String = "",
	val contract: String = "",
	val startTime: Calendar? = null,
	val endTime: Calendar? = null,
	val sinkMode: Boolean = false,
	val boostOrder: Int = BOOST_ORDER_UNKNOWN,

	val players: List<String> = emptyList(),
	val sink: String = "",

	/** Used to lock a players into certain boost order positions. */
	val playerPositionOverrides: Map<String, Int> = emptyMap(),

	/** Used to have certain players following a different boost order than the default. */
	val playerOrderOverrides: Map<String, Int> = emptyMap(),

	/** Overrides the number of tokens each player wants. */
	val playerTokenAmounts: Map<String, Int> = emptyMap()
) {
	companion object {
		// TODO: Should these be moved to another class?
		const val BOOST_ORDER_UNKNOWN = 0
		const val BOOST_ORDER_LUCK = 1
		const val BOOST_ORDER_EB = 2
	}
}
